torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -17,7 +16,7 @@
 
 # TODO: naming!
 def wrap_dataset_for_transforms_v2(dataset):
-    return VisionDatasetDatapointWrapper(dataset)
+    return type(f"Wrapped{type(dataset).__name__}", (VisionDatasetDatapointWrapper, type(dataset)), {})(dataset)
```

Comment:
This is the minimal change to get the `isinstance` check working without touching the original dataset code at all. I feel this is a little more hacky than the previous meta class approach, but this may be justified for this (hopefully) temporary wrapper.

We dynamically create a class here and instantiate it. For example, if the input is a `CocoDetection`, this is equivalent to

```python
class WrappedCocoDetection(VisionDatasetDatapointWrapper, CocoDetection):
    pass
```

Due to the double inheritance, `isinstance(dataset, CocoDetection)` now passes. However, we still use composition internally, since we hit the constructor of `VisionDatasetDatapointWrapper` and never call `super().__init__`.